### PR TITLE
feat(EMS-2211): No PDF - Alternative trading address - Save and back

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction :pencil2:
This PR adds save and back functionality to the alternative trading address page

## Resolution :heavy_check_mark:
- Added route and constant for save and back
- Added controller for save and back for alternative trading address
- Added mapErrors function to map for different fieldId for alternative trading address input
- Added tests
- Changed to use FULL_ADDRESS as the fieldId on the alternative trading address page

## Miscellaneous :heavy_plus_sign:
- Renamed mock to `mockCompanyDifferentTradingAddress`

### CODE REVIEW

File: src/ui/server/controllers/insurance/business/map-submitted-data/company-different-trading-address/index.ts [hunk 1]
```diff
@@ -23,4 +23,21 @@ const mapSubmittedData = (formBody: RequestBody): object => {
   return {};
 };
 
-export default mapSubmittedData;
+/**
+ * maps validation errors for companyDifferentTradingAddress
+ * Adds object in errorList for FULL_ADDRESS if the error exists for ALTERNATIVE_TRADING_ADDRESS
+ * Allows for datasaving as field is stored with different name as compared to form
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/map-submitted-data/company-different-trading-address/index.ts:29]: `datasaving`

File: src/ui/server/controllers/insurance/business/map-and-save/company-different-trading-address/index.ts [hunk 2]
```diff
@@ -15,11 +15,17 @@ const companyDifferentTradingAddress = async (formBody: RequestBody, application
   try {
     if (hasFormData(formBody)) {
       const dataToSave = mapSubmittedData(formBody);
+      /**
+       * maps errors for companyDifferentAddress
+       * Adds errors for fullAddress to validationErrors object
+       * As field is fullAddress in database
+       */
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/map-and-save/company-different-trading-address/index.ts:22]: I think this comment could just be 1 line or removed completely, as `mapErrors` has documentation

I just thought - should we have the field ID in the UI called `FULL_ADDRESS`? Would avoid this mapping/reduce complexity

File: src/ui/server/controllers/insurance/business/map-submitted-data/company-different-trading-address/index.ts (hunk 1 above)
Comment (Zainzzkk) [reply src/ui/server/controllers/insurance/business/map-submitted-data/company-different-trading-address/index.ts:29]: removed

File: src/ui/server/controllers/insurance/business/map-and-save/company-different-trading-address/index.ts (hunk 2 above)
Comment (Zainzzkk) [reply src/ui/server/controllers/insurance/business/map-and-save/company-different-trading-address/index.ts:22]: changed to FULL_ADDRESS

File: e2e-tests/insurance/cypress/e2e/journeys/your-business/alternative-trading-address/alternative-trading-address.spec.js [hunk 3]
```diff
@@ -70,9 +70,9 @@ context('Insurance - Your business - Alternative trading address page - I want t
     cy.saveSession();
   });
 
-  after(() => {
-    cy.deleteApplication(referenceNumber);
-  });
+  // after(() => {
+  //   cy.deleteApplication(referenceNumber);
+  // });
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-business/alternative-trading-address/alternative-trading-address.spec.js:75]: ?

File: e2e-tests/insurance/cypress/e2e/journeys/your-business/alternative-trading-address/save-and-back.spec.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+const { MAXIMUM } = FIELDS[ALTERNATIVE_TRADING_ADDRESS];
+
+const { STATUS: { IN_PROGRESS } } = TASKS;
+
+const task = partials.taskList.prepareApplication.tasks.business;
+
+const baseUrl = Cypress.config('baseUrl');
+
+const { DIFFERENT_TRADING_ADDRESS } = application;
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-business/alternative-trading-address/save-and-back.spec.js:30]: should this be `FULL_APPLICATION` now?

File: src/ui/server/controllers/insurance/business/alternative-trading-address/index.test.ts [hunk 5]
```diff
@@ -125,21 +125,22 @@ describe('controllers/insurance/business/alternative-trading-address', () => {
...
+          SAVE_AND_BACK_URL: `${INSURANCE_ROOT}/${referenceNumber}${SAVE_AND_BACK}`,
         });
       });
     });
 
     describe('when there are no validation errors', () => {
       beforeEach(() => {
         req.body = {
-          [ALTERNATIVE_TRADING_ADDRESS]: 'test',
+          [FULL_ADDRESS]: 'test',
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/alternative-trading-address/index.test.ts:136]: Not a blocker - more descriptive mock value?

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**